build.sbt

```diff
@@ -271,6 +274,21 @@ lazy val connectCommon = (project in file("spark-connect/common"))
       PB.gens.java -> (Compile / sourceManaged).value,
       PB.gens.plugin("grpc-java") -> (Compile / sourceManaged).value
     ),
+    (assembly / test) := { },
```

Comment:
Hey @scottsand-db, I just have two quick questions, 

1. We are shading the Delta Connect projects, and the Delta Connect projects which only compiles against Spark Master, do we need to apply `runTaskOnlyOnSparkMaster` to Delta Connect's `assembly` as well?

2. If so, the `assembly` task is of type `sbt.TaskKey[sbt.File]` which expects the `emptyValue` of `runTaskOnlyOnSparkMaster` to be an empty file, can we just create a temporary dummy empty file?

`def createEmptyFile(): File = {
  val file = Files.createTempFile("empty", ".tmp").toFile
  file.deleteOnExit() // Ensure the file is deleted on exit
  file
}`

Reply:
Well, what is the current behavior now? Is something broken or not working as expected that is prompting you to add more code / change behavior?

Reply:
@scottsand-db 

1. I observed that with `build/sbt clean connectClient/assembly`, even though `assembly` depends on `compile`, so `assembly` should be skipped because we are not using Spark Master here, it's still compile and assemble some jars, it should be skipped like `compile` and `test` also.

But I think the reason why we introduced `runTaskOnlyOnSparkMaster` back in the days is we want `build/sbt compile` and `build/sbt test` to work right? Do we also need `build/sbt assembly` to work? I am assuming if we didn't introduce it back then, then we don't need to introduce `runTaskOnlyOnSparkMaster` for `assembly` right?

2. My understanding is that when we release new version of Delta, we only do `build/sbt publish` and upload all the published jars of the projects to here right? (https://oss.sonatype.org/content/repositories/iodelta-1166/io/delta/) You guys don't use the `assembly` command right?

Then I guess my goal here is to make sure `build/sbt publish` published the shaded jars of the Delta Connect projects when we reach Delta 4.0.

I saw that back in the days we had to deal with the `standalone` project in [a special way](https://github.com/delta-io/delta/blame/master/build.sbt#L1144), I also saw that we are also having 2 versions of [`iceberg`](https://github.com/delta-io/delta/blame/master/build.sbt#L834), is there like some special treatment we have to do to publish shaded jars of projects?

Reply:
> it should be skipped like compile and test also.

Sounds good. Let's implement this. do you need help with this?

Reply:
> Then I guess my goal here is to make sure build/sbt publish published the shaded jars of the Delta Connect projects when we reach Delta 4.0.

> I saw that back in the days we had to deal with the standalone project in a special way, I also saw that we are also having 2 versions of iceberg, is there like some special treatment we have to do to publish shaded jars of projects?

I think your code is missing `Compile / packageBin := assembly.value,`

You can easily test this, too. Just run `build/sbt publishM2` and verify that the delta connect artifacts contain what you want them to.

Let me know if you need help with this also.

Reply:
@scottsand-db 

1. "Sounds good. Let's implement this. do you need help with this?" Yes please, that would be great, thank you so much Scott!

2. "I think your code is missing Compile / packageBin := assembly.value,

You can easily test this, too. Just run build/sbt publishM2 and verify that the delta connect artifacts contain what you want them to."

Cool, looks like it works, I just run `build/sbt clean -DsparkVersion=master "project connectClient" publishM2` and then use `jar tf` to manually inspect my jar file `long.vu/.m2/repository/io/delta/delta-connect-client_2.13/3.3.0-SNAPSHOT/delta-connect-client_2.13-3.3.0-SNAPSHOT.jar`

Reply:
Hey @scottsand-db, do you have any updates on adding `runTaskOnlyOnSparkMaster` to `assembly`? Is it more difficult than the `publish`, `compile` and `test`?

Reply:
```scala
assembly := runTaskOnlyOnSparkMaster(
  task = assembly,
  taskName = "assembly",
  projectName = "delta-connect-common",
  emptyValue = file("")
).value,
```

Reply:
@scottsand-db Thank you very much for your help and addressing my original question! I didn't see that `file("")` was a thing, it is much more cleaner and neat than my original solution.